packages/visx-xychart/src/hooks/useStackedData.ts

```diff
@@ -0,0 +1,96 @@
+import React, { useContext, useEffect, useMemo } from 'react';
+import { SeriesPoint, stack as d3stack } from 'd3-shape';
+import stackOffset from '@visx/shape/lib/util/stackOffset';
+import stackOrder from '@visx/shape/lib/util/stackOrder';
+import { StackPathConfig } from '@visx/shape';
+import { extent } from 'd3-array';
+import { AxisScale } from '@visx/axis';
+import DataContext from '../context/DataContext';
+import { CombinedStackData, DataContextType, SeriesProps } from '../types';
+import getBarStackRegistryData from '../utils/getBarStackRegistryData';
+import combineBarStackData from '../utils/combineBarStackData';
+import isChildWithProps from '../typeguards/isChildWithProps';
+
+type UseStackedData<Datum extends object> = {
+  children: JSX.Element | JSX.Element[];
+} & Pick<StackPathConfig<Datum, string>, 'offset' | 'order'>;
+
+export default function useStackedData<
+  XScale extends AxisScale,
+  YScale extends AxisScale,
+  Datum extends object,
+  ChildrenProps extends SeriesProps<XScale, YScale, Datum>
+>({ children, order, offset }: UseStackedData<Datum>) {
+  type StackDatum = SeriesPoint<CombinedStackData<XScale, YScale>>;
+
+  const { horizontal, registerData, unregisterData } = (useContext(
+    DataContext,
+  ) as unknown) as DataContextType<XScale, YScale, StackDatum>;
+
+  // find series children
+  // @TODO: memoization doesn't work well if at all for this
```

Comment:
what do you mean by doesn't work well? still slow?

Reply:
or is it because the equality check doesn't work well for `children`

Reply:
equality check doesn't work with `children`, so I think basically everything in this hook is **always** re-run. I can probably optimize it with `ref`s to the child data arrays (which would still break if users have inline data but hopefully not).

I plan to follow up on this in another PR with tests for it.